buildtest/builders/base.py

```diff
@@ -932,17 +932,15 @@ def _get_variables(self, variables):
 
         return lines
 
-    def _extract_content(self, regex):
-        """Extract content based on the stream and linenum properties
-        from the regex field and return it as a string.
+    def _extract_content(self, regex, content):
+        """Extract content based on the linenum property from the regex or file_regex
+        field and return it as a string.
 
         Args:
-            regex (dict): regex section from the metrics field
+            regex (dict): regex or file_regex section from the metrics field
+            content (str): content to be extracted from
         """
 
-        stream = regex.get("stream")
-        content = self._output if stream == "stdout" else self._error
-
         linenum = regex.get("linenum")
```

Comment:
i am thinking since `regex` is used once just to check this assignment`linenum = regex.get('linenum')`, why not we do this outside the method. IF the `linenum` property is not defined we just need to set to empty quotes.

Reply:
ah, I am confused again. If you just pass in content_input, how do you know which linenum to extract.
I must be missing something important.

Reply:
Also, it regex.get("linenum") is not None, content = content_input instead of ""

Reply:
okay so please take a look at https://github.com/buildtesters/buildtest/pull/1751/files#r1572440374 to help address your confusion. I think we do need 2 arguments, `linenum` and `content` and not `regex`.

Reply:
👍